5.0/debian-10/rootfs/opt/bitnami/scripts/redis-cluster/run.sh

```diff
@@ -16,24 +19,33 @@ set -o pipefail
 
 IFS=' ' read -ra nodes <<< "$REDIS_NODES"
 
-if ! is_boolean_yes "$REDIS_CLUSTER_CREATOR"; then
-    ARGS=("--port" "$REDIS_PORT_NUMBER")
-    ARGS+=("--include" "${REDIS_BASE_DIR}/etc/redis.conf")
+ARGS=("--port" "$REDIS_PORT_NUMBER")
+ARGS+=("--include" "${REDIS_BASE_DIR}/etc/redis.conf")
 
-    if ! is_boolean_yes "$ALLOW_EMPTY_PASSWORD"; then
-        ARGS+=("--requirepass" "$REDIS_PASSWORD")
-        ARGS+=("--masterauth" "$REDIS_PASSWORD")
-    else
-        ARGS+=("--protected-mode" "no")
-    fi
+if ! is_boolean_yes "$ALLOW_EMPTY_PASSWORD"; then
+    ARGS+=("--requirepass" "$REDIS_PASSWORD")
+    ARGS+=("--masterauth" "$REDIS_PASSWORD")
+else
+    ARGS+=("--protected-mode" "no")
+fi
 
-    ARGS+=("$@")
+ARGS+=("$@")
 
+if ! is_boolean_yes "$REDIS_CLUSTER_CREATOR"; then
     if am_i_root; then
         exec gosu "$REDIS_DAEMON_USER" redis-server "${ARGS[@]}"
     else
         exec redis-server "${ARGS[@]}"
     fi
 else
+    # Start Redis in background
+    if am_i_root; then
+        gosu "$REDIS_DAEMON_USER" redis-server "${ARGS[@]}" &
+    else
+        redis-server "${ARGS[@]}" &
+    fi
+    # Create the cluster
     redis_cluster_create "${nodes[@]}"
+    # Bring redis process to foreground
+    fg
 fi
```

Comment:
It's unnatural to me to use `if ! (condition) ... else .... fi)`. Change the order to avoid negating the condition:
```suggestion
if is_boolean_yes "$REDIS_CLUSTER_CREATOR"; then
    # Start Redis in background
    if am_i_root; then
        gosu "$REDIS_DAEMON_USER" redis-server "${ARGS[@]}" &
    else
        redis-server "${ARGS[@]}" &
    fi
    # Create the cluster
    redis_cluster_create "${nodes[@]}"
    # Bring redis process to foreground
    fg
else
    if am_i_root; then
        exec gosu "$REDIS_DAEMON_USER" redis-server "${ARGS[@]}"
    else
        exec redis-server "${ARGS[@]}"
    fi
fi
```